frontend/platform.bible-extension/src/main.ts

```diff
@@ -1,80 +1,264 @@
+  ): Promise<WebViewDefinition | undefined> {
+    if (savedWebView.webViewType !== findWordWebViewType)
       throw new Error(
-        `${reactWebViewType} provider received request to provide a ${savedWebView.webViewType} web view`,
+        `${findWordWebViewType} provider received request to provide a ${savedWebView.webViewType} web view`,
       );
     return {
       ...savedWebView,
-      title: 'FW Lite Extension React',
-      content: extensionTemplateReact,
-      styles: extensionTemplatestyles,
-      iconUrl: 'papi-extension://fw-lite-extension/assets/logo-dark.png',
-      allowedFrameSources: ['http://localhost:*']
+      ...options,
+      content: fwFindWordWindow,
+      iconUrl,
+      title: '%fwLiteExtension_findWord_title%',
     };
   },
 };
 
-export async function activate(context: ExecutionActivationContext) {
-  logger.info('FwLite is activating!');
+export async function activate(context: ExecutionActivationContext): Promise<void> {
+  logger.info('FieldWorks Lite is activating!');
 
-  const reactWebViewProviderPromise = papi.webViewProviders.register(
-    reactWebViewType,
-    reactWebViewProvider,
+  const mainWebViewProviderPromise = papi.webViewProviders.registerWebViewProvider(
+    mainWebViewType,
+    mainWebViewProvider,
+  );
+  const addWordWebViewProviderPromise = papi.webViewProviders.registerWebViewProvider(
+    addWordWebViewType,
+    addWordWebViewProvider,
+  );
+  const dictionarySelectWebViewProviderPromise = papi.webViewProviders.registerWebViewProvider(
+    dictionarySelectWebViewType,
+    dictionarySelectWebViewProvider,
+  );
+  const findWordWebViewProviderPromise = papi.webViewProviders.registerWebViewProvider(
+    findWordWebViewType,
+    findWordWebViewProvider,
   );
 
-  // Emitter to tell subscribers how many times we have done stuff
   const onFindEntryEmitter = papi.network.createNetworkEventEmitter<FindEntryEvent>(
-    'fwLiteExtension.findEntry',
+    'fwLiteExtension.findEntryEvent',
+  );
+
+  const urlHolder: UrlHolder = { baseUrl: '', dictionaryUrl: '' };
+  const { fwLiteProcess, baseUrl } = launchFwLiteFwLiteWeb(context);
+  urlHolder.baseUrl = baseUrl;
+
+  const fwLiteApi = new FwLiteApi(baseUrl);
+
+  const entryService = papi.networkObjects.set(
+    'fwliteextension.entryService',
+    new EntryService(baseUrl),
+    'fw-lite-extension.IEntryService',
   );
-  const onLaunchServerEmitter = papi.network.createNetworkEventEmitter<LaunchServerEvent>(
-    'fwLiteExtension.launchServer',
+
+  const validDictionaryCode = papi.projectSettings.registerValidator(
+    'fw-lite-extension.fwDictionaryCode',
+    async (dictionaryCode) => {
+      if (!dictionaryCode) {
+        logger.info('FieldWorks dictionary code cleared in project settings');
+        return true;
+      }
+      logger.info('Validating FieldWorks dictionary code:', dictionaryCode);
+      try {
+        return !!(await fwLiteApi.getWritingSystems(dictionaryCode)).analysis;
```

Comment:
we could make an api like `getProject(code: string)` which could be used for this purpose. It would also give you the project type so you can avoid hardcoding it.